dts/xtensa/espressif/esp32.dtsi

```diff
@@ -31,11 +31,27 @@
 	};
 
 	soc {
-		uart0: uart@40008fd0 {
-			compatible = "espressif,esp32-uart";
-			reg = <0x40008fd0 0x400>;
-			label = "ROMUART";
+		uart0: uart@3ff40000 {
+			compatible = "xtensa,esp32-uart";
```

Comment:
I think we should keep "espressif" as vendor. "xtensa" is not exactly a vendor.

Reply:
> I think we should keep "espressif" as vendor. "xtensa" is not exactly a vendor.

Is the IP for the UART coming from 'xtensa' or 'espressif'?

Reply:
@galak @dcpleung 

I think it is not easy to be 100% sure if IP comes with Xtensa or made by espressif, i tried to do a quick search but there is no solid info about it AFAIK.

i think the fix would be to rename file `xtensa,esp32-uart.yaml` to `espressif,esp32-uart.yaml` because `xtensa` is not a vendor and also to keep naming consistent in `dts/binding/`.
then update the `compatible` value in the dts.

feel free to suggest alternatives

Reply:
the IP is owned by espressif. Using xtensa here is wrong if we are looking for a vendor, xtensa is not a vendor (Cadence is)

Reply:
so, the correct naming is `cadence,esp32-uart` or `cadence,espressif-esp32-uart` ?

Reply:
I think it should be espressif. They own the IP, not cadence.

Reply:
I agree with @nashif. It should be `espressif,esp32-uart` and `espressif,esp32-uart.yaml`. Cadence owns Tensilica Xtensa processors, such as the Xtensa LX6 dataplane processing units (DPUs).